src/sentry/services/hybrid_cloud/user.py

```diff
@@ -0,0 +1,147 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from inspect import signature
+from typing import Any, Iterable, List, Optional, Sequence, Union
+
+from sentry.models.group import Group
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.user import User
+from sentry.services.hybrid_cloud import (
+    CreateStubFromBase,
+    InterfaceWithLifecycle,
+    silo_mode_delegation,
+)
+from sentry.silo import SiloMode
+
+
+@dataclass(frozen=True)
+class APIUser:
+    id: int = -1
+    pk: int = -1
+    name: str = ""
+    email: str = ""
+    username: str = ""
+    actor_id: int = -1
+
+    # TODO: Extract these "view" methods to BaseUser class?
+    # BaseUser class would also be useful for catching `isinstance(foo, User)` checks
+    # which may no longer mean what they used to mean
+    def class_name(self):
+        """Necessary for integrations that introspect on user objects"""
+        return "User"
+
+    def get_display_name(self):
```

Comment:
Does it make sense for this to just be a static property on APIUser?  And let the control silo pack it in with its own logic?

Thing is, I can see API drift causing issues with this.  We deploy control silo and it shows one name for users, but it's not in agreement with regions until they are all fully deployed.

If Control silo is source of truth for these sorts of "computed properties", I'd argue that at some point they should be static properties and determined as their source only.  I'm ok with this for now, but I'd atleast put a huge "KILL ME" there.

Reply:
Having these computed properties come from Control Silo and be pat of the User 'dataclass' seems like a good direction to go.